utils/self-extr-exec/compressor.cpp

```diff
@@ -297,7 +297,7 @@ int copy_decompressor(const char *self, int output_fd)
         return 1;
     }
 
-    char buf[1ul<<19];
+    char * buf = new char[1ul<<19];
```

Comment:
Since we decided to go C style, I think malloc and free are more appropriate here...